.github/workflows/main.yml

```diff
@@ -280,3 +280,24 @@ jobs:
         with:
           path-to-profile: coverage.txt
           parallel: true
+
+  ########################
+  # check pinned dependencies
+  ########################
+  dep-pin:
+    name: check pinned dependencies
+    runs-on: ubuntu-latest
+    strategy:
+      # Allow other tests in the matrix to continue if one fails.
+      fail-fast: false
+      matrix:
+        pinned_dep:
+          - google.golang.org/grpc v1.24.0
+          - github.com/golang/protobuf v1.3.2
+
+    steps:
+      - name: git checkout
+        uses: actions/checkout@v2
+
+      - name: ensure dependences at correct version
+        run: if [ -z "$(cat go.mod | grep "${{ matrix.pinned_dep }}")" ]; then echo dependency ${{ matrix.pinned_dep }} should not be altered ; exit 1 ; fi
```

Comment:
I think the if can be simplified to: `if ! grep -q "${{ matrix.pinned_dep }}" go.mod; then ...`.
The `-q` instructs grep to just return a 0 exit code if there was a match.